app/client/src/components/shared/AlertMessage/index.js

```diff
@@ -0,0 +1,107 @@
+// @flow
+
+import React from 'react';
+import styled from 'styled-components';
+import { globalHistory } from '@reach/router';
+// styles
+import { colors } from 'styles/index.js';
+// contexts
+import { useNotificationsContext } from 'contexts/LookupFiles';
+
+// --- components ---
+type Props = {};
+
+// --- styled components ---
+const Banner = styled.div`
+  background-color: ${(props) => props.backgroundColor};
+  color: ${(props) => props.color};
+  width: 100%;
+  margin: 0 auto;
+  padding: 10px 5px;
+  text-align: center;
+
+  p {
+    padding: 0;
+    margin: 0;
+  }
+`;
+
+const Separator = styled.hr`
+  margin: 0;
+  background-color: ${colors.gray9};
+`;
+
+function AlertMessage({ ...props }: Props) {
+  const notifications = useNotificationsContext();
+
+  const [pathname, setPathname] = React.useState('');
+
+  if (notifications.status === 'failure') return null;
+
+  if (notifications.status === 'fetching') return null;
+
+  if (
+    notifications.status === 'success' &&
+    (!notifications.data || Object.keys(notifications.data).length === 0)
+  ) {
+    return null;
+  }
+
+  if (
+    notifications.status === 'success' &&
+    Object.keys(notifications.data).length > 0
+  ) {
+    const data = notifications.data;
+
+    if (pathname !== window.location.pathname) {
+      setPathname(window.location.pathname);
+    }
+
+    // watch for history changes that wouldn't trigger a re-render of this component
+    globalHistory.listen(() => {
+      if (pathname !== window.location.pathname) {
+        setPathname(window.location.pathname);
+      }
+    });
+
+    function getPageFromPathname(pathname) {
+      return pathname.split('/')[1];
+    }
+
+    function createMarkup(message) {
+      return { __html: message };
+    }
```

Comment:
You could git rid of the code smell, that was flagged by sonarcloud, by moving these functions outside of the AlertMessage function component.

Reply:
Thanks! Because getPageFromPathname() is only called once I've removed this function. createMarkup() is used in the National NARS JSON branch so I've moved it to the utils file.